src/EventListener/ShippingExportEventListener.php

```diff
@@ -14,35 +14,64 @@
 
 use BitBag\SyliusDhl24PlShippingExportPlugin\Api\SoapClientInterface;
 use BitBag\SyliusDhl24PlShippingExportPlugin\Api\WebClientInterface;
-use BitBag\SyliusShippingExportPlugin\Event\ExportShipmentEvent;
+use BitBag\SyliusShippingExportPlugin\Entity\ShippingExportInterface;
+use Doctrine\Persistence\ObjectManager;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Webmozart\Assert\Assert;
 
 final class ShippingExportEventListener
 {
-    const DHL_GATEWAY_CODE = 'dhl24_pl';
-    const BASE_LABEL_EXTENSION = 'pdf';
+    /** @var FlashBagInterface */
+    private $flashBag;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    /** @var ObjectManager */
+    private $shippingExportManager;
+
+    /** @var string */
+    private $shippingLabelsPath;
 
     /** @var WebClientInterface */
     private $webClient;
 
     /** @var SoapClientInterface */
     private $soapClient;
 
-    public function __construct(WebClientInterface $webClient, SoapClientInterface $soapClient)
-    {
+    public function __construct(
+        FlashBagInterface $flashBag,
+        Filesystem $filesystem,
+        ObjectManager $shippingExportManager,
+        string $shippingLabelsPath,
+        WebClientInterface $webClient,
+        SoapClientInterface $soapClient
```

Comment:
Is it possible to break this listener into separate classes? 6 dependencies is a pretty high number. Once you try to write a unit test for this class, it'll be intuitive.